packages/components/src/components/assay/AssayImportPanels.tsx

```diff
@@ -272,7 +283,7 @@ class AssayImportPanelsImpl extends React.Component<Props, State> {
             // samples.
             this.props.loadSelections(location, sampleColumnData.column).then(samples => {
                 // Only one sample can be added at batch or run level, so ignore selected samples if multiple are selected.
-                let runProperties = this.getRunPropertiesRow(this.props);
+                let runProperties = this.populateAssayRequest(this.getRunPropertiesRow(this.props));
```

Comment:
I think I would probably put the call to `populateAssayRequest` inside `getRunPropertiesRow`.  Then we're assured it is always included. If any of the other calls to this method requires that the property not be included, you could add an optional `includeAssayRequest` parameter to `getRunPropertiesRow`.

Reply:
My first pass at this was to inject it at that level, but I don't believe appropriate for a few reasons:

- We only want it in the two contexts that it is currently used, all other contexts do not need it as far as I can tell, and there are many usages of getRunPropertiesRow.
- The value of the data in side the function is in most cases undefined which adds a layer of complexity by having to dance around potential undefined
- Injecting it in getRunPropertiesRow also requires that I handle the flatten case, or at least dance around it. I only inject it when we are flattening the data.
- In short: the implementation I have is more narrow, and requires less logic/checking to implement.